### PR TITLE
Add test to torch.jit.export_opnames, make the _C function private

### PR DESCRIPTION
**Pull request by iseeyuan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31446 Add test to torch.jit.export_opnames, make the _C function private**

Differential Revision: [D19172851](https://our.internmc.facebook.com/intern/diff/D19172851)

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit e15ff5d:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=e15ff5dd0d67661f6d05705e629d7c892232a08f).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4009064) pytorch_macos_10_13_py3_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=77002962) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=77002962))
<details>
<summary>
<code>Dec 18 15:09:09 RuntimeError: test_docs_coverage failed!</code>
</summary>

```
Dec 18 15:09:09  
Dec 18 15:09:09 ---------------------------------------------------------------------- 
Dec 18 15:09:09 Ran 2 tests in 0.004s 
Dec 18 15:09:09  
Dec 18 15:09:09 FAILED (failures=1) 
Dec 18 15:09:09 Traceback (most recent call last): 
Dec 18 15:09:09   File "test/run_test.py", line 456, in <module> 
Dec 18 15:09:09     main() 
Dec 18 15:09:09   File "test/run_test.py", line 449, in main 
Dec 18 15:09:09     raise RuntimeError(message) 
Dec 18 15:09:09 RuntimeError: test_docs_coverage failed! 
Dec 18 15:09:09 + cleanup 
Dec 18 15:09:09 + retcode=1 
Dec 18 15:09:09 + set +x 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Review by ZolotukhinM — APPROVED** [review]
Looks good to me, but feel free to wait for other reviewers too.

File: test/test_jit.py [hunk 1]
```diff
@@ -3706,6 +3706,21 @@ def lstm(x, hx, cx, w_ih, w_hh, b_ih, b_hh):
...
+                def __init__(self):
+                    super(Foo, self).__init__()
+
+                @torch.jit.script_method
+                def forward(self, x, y):
+                    return 2 * x + y
+
+            foo = Foo()
+            ops = torch.jit.export_opnames(foo)
+            expected = ['aten::add.Tensor', 'aten::mul.Scalar', 'prim::Constant']
```
[the remark this answers is not in the document]
Comment (iseeyuan) [reply test/test_jit.py]: Yes, the output is in alphabet order (sorted inside export_opnames). You suggested that the order should be static in the original PR.

[the remark this answers is not in the document]
Comment (suo) [reply test/test_jit.py]: Can we make some more complex test cases as well? I would like to see, at least:
1. Submodules
2. Methods that call other methods
3. Submodules that are interfaces

Just to make sure we're not missing obvious cases.

File: test/test_jit.py [hunk 2]
```diff
@@ -3693,6 +3693,21 @@ def lstm(x, hx, cx, w_ih, w_hh, b_ih, b_hh):
                 self.assertTrue(type(block.paramNode()) == torch._C.Node)
         self.assertTrue(tested_blocks)
 
+    def test_export_opnames(self):
+        with torch.jit._disable_emit_hooks():
```
Comment (suo) [inline test/test_jit.py]: Why is this used?

Comment (iseeyuan) [reply test/test_jit.py]: It's a copy-paste. Removed for the updated PR.

File: test/test_jit.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (iseeyuan) [reply test/test_jit.py]: As discussed offline, I've added submodules and methods that call other methods, to go into 1.4.0. I'll cover interfaces in another PR.

File: test/test_jit.py [hunk 3]
```diff
@@ -17486,7 +17517,7 @@ def test_docs(self):
     add_nn_module_test(**test)
 
 if __name__ == '__main__':
-    run_tests()
+    # run_tests()
```
Comment (gchanan) [inline test/test_jit.py]: did you mean to include this?

**Review by gchanan — APPROVED** [review]

File: test/test_jit.py (hunk 3 above)
Comment (iseeyuan) [reply test/test_jit.py]: It's for quick debug. Just added it back. Thanks!

**Pull request closed without merging**